contract/AElf.Contracts.Profit/ProfitContract.cs

```diff
@@ -280,134 +307,197 @@ public override Empty SubWeights(SubWeightsInput input)
         }
 
         /// <summary>
-        /// Higher level profit item has already released.
-        /// Otherwise this profit item maybe has nothing to release.
-        /// This pre-condition must be met before calling this method.
+        /// Will burn/destroy a certain amount of profits if `input.Period` is less than 0.
         /// </summary>
         /// <param name="input"></param>
         /// <returns></returns>
         public override Empty ReleaseProfit(ReleaseProfitInput input)
         {
-            Assert(input.Amount >= 0, $"Amount must greater than 0");
-            var profitItem = State.ProfitItemsMap[input.ProfitId];
+            Assert(input.Amount >= 0, "Amount must be greater than or equal to 0");
+
+            Assert(input.Symbol != null && input.Symbol.Any(), "Invalid token symbol.");
+            if (input.Symbol == null) return new Empty(); // Just to avoid IDE warning.
 
+            var profitItem = State.ProfitItemsMap[input.ProfitId];
             Assert(profitItem != null, "Profit item not found.");
+            if (profitItem == null) return new Empty(); // Just to avoid IDE warning.
 
-            if (profitItem == null)
+            Assert(Context.Sender == profitItem.Creator, "Only creator can release profits.");
+
+            var balance = AssertBalanceIsEnough(profitItem.VirtualAddress, input);
```

Comment:
No need to call another contract to assert something that rarely happens. It's too expensive and not necessary. The  txn fails no matter you assert or not.